src/core/src/CoreMain.py

```diff
@@ -41,16 +42,17 @@ def __init__(self, argv):
             lifecycle_manager, status_handler = bootstrapper.build_core_components(container)
             composite_logger.log_debug("Completed building out full container.\n\n")
 
-            # Basic environment check
-            bootstrapper.bootstrap_splash_text()
-            bootstrapper.basic_environment_health_check()
-            lifecycle_manager.execution_start_check()      # terminates if this instance shouldn't be running (redundant)
-
             # Execution config retrieval
             composite_logger.log_debug("Obtaining execution configuration...")
             execution_config = container.get('execution_config')
             telemetry_writer.set_operation_id(execution_config.activity_id)
             patch_operation_requested = execution_config.operation.lower()
+
+            # Basic environment check
+            bootstrapper.bootstrap_splash_text()
+            bootstrapper.basic_environment_health_check()
+            lifecycle_manager.execution_start_check()  # terminates if this instance shouldn't be running (redundant)
```

Comment:
Bootstrapper splash needs to be first. Execution start check also needs to come immediately after - we don't want/need telemetry from there to be associated with activity id. 
- Don't want because it'd have spurious entries associated with activity id and we also want the least amount of code to run prior to stopping. 
- Don't need because the telemetry is associated with the machine and can be retrieved linearly in case it's needed.

The basic environment health check can come later after execution config retrieval... You'd need to section and comment the sections appropriately.